config/v1/types_infrastructure.go

```diff
@@ -610,6 +621,15 @@ type GCPPlatformStatus struct {
 	// +optional
 	// +openshift:enable:FeatureSets=CustomNoUpgrade;TechPreviewNoUpgrade
 	ResourceTags []GCPResourceTag `json:"resourceTags,omitempty"`
+
+	// ClusterHostedDNS indicates the type of DNS being used by the cluster. When Enabled the cluster runs a DNS
```

Comment:
I would like to see this explain why/when I would expect this DNS service to operate, and what is using the service. At the moment this is very generic and a user might expect to be able to configure their own DNS entries, which I don't think is the intention here.

Also, godoc should start with json tag name

Reply:
```
// clusterHostedDNS indicates the type of DNS solution in use within the cluster. Its default value of 
// "Disabled" indicates that the cluster's DNS is the default provided by the cloud platform. It can be 
// "Enabled" during install to bypass the configuration of the cloud default DNS. When "Enabled", the
// cluster needs to provide a self-hosted DNS solution for the cluster's installation to succeed. 
// The cluster's use of the cloud's Load Balancers is unaffected by this setting.
// Once set at install time, changing its value will not have any effect on the cluster's DNS.
// Currently, there is no way for the customer to add additional DNS entries into the cluster hosted DNS.
// Enabling this functionality allows the user to start their own DNS solution outside the cluster after
// installation is complete. The customer would be responsible for configuring this custom DNS solution
// and it can be run in addition to the in-cluster DNS solution.
```

@JoelSpeed, @barbacbd does this provide better context for this field and its use?

Reply:
I like it but is it too verbose? It is certainly information that should be provided to the openshift docs. If this isn't too verbose I can gladly add this to the PR.